src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -100,13 +98,14 @@ class NexusPublishPlugin : Plugin<Project> {
         }
         rootProject.tasks.withType(AbstractTransitionNexusStagingRepositoryTask::class.java).configureEach {
             transitionCheckOptions.convention(extension.transitionCheckOptions)
-            stagingRepositoryId.convention(registry.map { it[repository.get().name].stagingRepositoryId })
+            usesService(registryService)
+            stagingRepositoryId.convention(registryService.map { it.registry[repository.get().name].stagingRepositoryId })
         }
         extension.repositories.all {
-            username.convention(rootProject.provider { rootProject.findProperty("${name}Username") as? String })
-            password.convention(rootProject.provider { rootProject.findProperty("${name}Password") as? String })
+            username.convention(rootProject.providers.gradleProperty("${name}Username").forUseAtConfigurationTimeCompat())
+            password.convention(rootProject.providers.gradleProperty("${name}Password").forUseAtConfigurationTimeCompat())
```

Comment:
Behavior change as discussed in https://github.com/gradle-nexus/publish-plugin/pull/242#discussion_r1257338630

> To be honest, I'm not sure about that chance. I'm ambivalent to it. What do you propose? - @szpak 

Assuming you're referring to the chance of someone using something other than `-P`: I think a major version change is the best time to introduce breaking changes like this. It won't come unexpected. And we can list it in a migration guide as something to look out for.

Reply:
... and if someone does use it, and want it, they'll hopefully tell us :)

Reply:
I was rather writing about benefits of bumping required version to 6.2 :-). Nevertheless, the newer API we use, the (hopefully) later it will be deprecated in Gradle :-).